### PR TITLE
implements NavigatorLanguage

### LINKED ISSUE #9992: Implement NavigatorLanguage
Spec: https://html.spec.whatwg.org/multipage/webappapis.html#navigatorlanguage (and add `Navigator implements NavigatorLanguage;`)
Code: `components/script/dom/webidls/Navigator.webidl`, `components/script/dom/navigator.rs`
Tests:
* `./mach test-wpt tests/wpt/html/dom/interfaces.html`
* `./mach test-wpt tests/wpt/web-platform-tests/html/webappapis/system-state-and-capabilities/the-navigator-object/navigatorlanguage.html`
* Follow [the documentation](https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations) to adjust for tests that now pass.

The lack of this feature is makes clicking links on Twitter break (see #9870). Let's just hardcode this as `en-US` right now.

### PR DESCRIPTION
**Pull request by creativcoder**
Fixes #9992

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10009)

### CODE REVIEW

**Comment by Ms2ger** [comment]
Tidy is a little grumpy:
```
./components/script/dom/navigator.rs:68: trailing whitespace
./components/script/dom/navigator.rs:69: link to WHATWG may break in the future, use this format instead: https://html.spec.whatwg.org/multipage/#navigatorlanguage
./components/script/dom/webidls/Navigator.webidl:29: link to WHATWG may break in the future, use this format instead: https://html.spec.whatwg.org/multipage/#navigatorlanguage
```

File: tests/wpt/metadata/html/dom/interfaces.html.ini [hunk 1]
```diff
@@ -7554,9 +7551,6 @@
   [SharedWorker interface: attribute onerror]
     expected: FAIL
 
-  [WorkerNavigator interface: attribute language]
```
Comment (jdm) [inline tests/wpt/metadata/html/dom/interfaces.html.ini]: I am surprised by this change.

Comment (creativcoder) [inline tests/wpt/metadata/html/dom/interfaces.html.ini]: Ah, i see. So this need not change here. Okay.

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Comment by Wafflespeanut** [comment]
▶ Unexpected subtest result in /html/dom/interfaces.html:
    └ PASS [expected FAIL] Navigator interface: window.navigator must inherit property "language" with the proper type (7)

    ▶ Unexpected subtest result in /workers/interfaces/WorkerUtils/navigator/language.html:
    │ FAIL [expected PASS] navigator.language
    │   → assert_equals: expected (string) "en-US" but got (undefined) undefined
    │ 
    │ worker.onmessage<@http://web-platform.test:8000/workers/interfaces/WorkerUtils/navigator/language.html:4:5
    │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
    └ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:1405:1

**Comment by creativcoder** [comment]
First test case, was resolved by removing the entry from interfaces.html.ini. But am not sure of what to do of this second failing case. If i add `language` to `WorkerNavigator` and when tested with devtools, `navigator.language`, returns "en-US", in worker's scope, but this still does not make this test pass.
```
▶ Unexpected subtest result in /workers/interfaces/WorkerUtils/navigator/language.html:
│ FAIL [expected PASS] navigator.language
│   → assert_equals: expected (string) "en-US" but got (undefined) undefined
│ 
│ worker.onmessage<@http://web-platform.test:8000/workers/interfaces/WorkerUtils/navigator/language.html:4:5
│ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
└ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:1405:1
```

**Comment by jdm** [comment]
That's odd; I can't think of any reason that would happen. Did you uncomment http://mxr.mozilla.org/servo/source/components/script/dom/webidls/WorkerNavigator.webidl#10 as well?

**Comment by creativcoder** [comment]
Yes, i did.

**Comment by KiChjang** [comment]
Does the test use postMessage? Looking at the test file, the comment at the top says postMessage(navigator.language).

**Comment by jdm** [comment]
Workers support postMessage.

File: components/script/dom/navigator.rs [hunk 2]
```diff
@@ -72,5 +72,9 @@ impl NavigatorMethods for Navigator {
     // https://webbluetoothcg.github.io/web-bluetooth/#dom-navigator-bluetooth
     fn Bluetooth(&self) -> Root<Bluetooth> {
         self.bluetooth.or_init(|| Bluetooth::new(self.global().r()))
+
```
Comment (jdm) [inline components/script/dom/navigator.rs]: Rebase error; the } is missing.

Comment (creativcoder) [inline components/script/dom/navigator.rs]: Ah, my bad. Fixing it. Btw, i got to know a lot, from this PR. :)

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] WorkerNavigator interface: attribute language

  ▶ Unexpected subtest result in /workers/interfaces.worker:
  └ PASS [expected FAIL] WorkerNavigator interface: attribute language

  ▶ Unexpected subtest result in /workers/interfaces.worker:
  └ PASS [expected FAIL] WorkerNavigator interface: self.navigator must inherit property "language" with the proper type (6)
```

**Comment by Ms2ger** [comment]
While we're having you jump through hoops, the spec changed `DOMString? language` to `DOMString language`, want to update that too?

**Comment by Ms2ger** [comment]
Thanks!

@bors-servo r+

**Pull request merged**